README.md

```diff
@@ -15,6 +47,507 @@ go get -u github.com/src-d/kallax/...
+
+### Update models
+
+To insert a model we just need to use the `Update` method of the store and pass it a model. It will return an error if the model was not already persisted or has not an ID.
+
+```go
+user := FindLast()
+rowsUpdated, err := store.Update(user)
+if err != nil {
+        // handle error
+}
+```
+
+By default, when a model is updated, all its fields are updated. You can also specify which fields to update passing them to update.
+
+```go
+rowsUpdated, err := store.Update(user, Schema.User.Username, Schema.User.Password)
+if err != nil {
+        // handle error
+}
+```
+
+If our model has relationships, they will be saved (**note:** saved as in insert or update) as well. The relationships of the relationships will not, though. Relationships are only saved with one level of depth.
+
+```go
+user := FindLastPoster()
+rowsUpdated, err := store.Update(user)
+if err != nil {
+        // handle error
+}
+```
+
+If there are any relationships in the model, both the model and the relationships will be saved in a transaction and only succeed if all of them are saved correctly.
+
+### Save models
+
+To save a model we just need to use the `Save` method of the store and pass it a model. It will update the model if it was already persisted or insert it otherwise.
+
+```go
+updated, err := store.Save(user)
+if err != nil {
+        // handle error
+}
+
+if updated {
+        // it was updated, not inserted
+}
+```
+
+If our model has relationships, they will be saved as well. The relationships of the relationships will not, though. Relationships are only saved with one level of depth.
+
+```go
+user := NewUser("foo")
+user.Posts = append(user.Posts, NewPost(user, "new post"))
+
+updated, err := store.Save(user)
+if err != nil {
+        // handle error
+}
+```
+
+If there are any relationships in the model, both the model and the relationships will be saved in a transaction and only succeed if all of them are saved correctly.
+
+### Delete models
+
+To delete a model we just have to use the `Delete` method of the store. It will return an error if the model was not already persisted.
+
+```go
+err := store.Delete(user)
+if err != nil {
+        // handle error
+}
+```
+
+Relationships of the model are **not** automatically removed using `Delete`.
+
+For that, specific methods are generated in the store of the model.
+
+For one to many relationships:
+
+```go
+// remove specific posts
+err := store.RemovePosts(user, post1, post2, post3)
+if err != nil {
+        // handle error
+}
+
+// remove all posts
+err := store.RemovePosts(user)
+```
+
+For one to one relationships:
+
+```go
+// remove the thing
+err := store.RemoveThing(user)
+```
+
+## Query models
+
+### Simple queries
+
+To perform a query you have to do the following things: 
+* Create a query
+* Pass the query to `Find`, `FindOne`, `MustFind` or `MustFindOne` of the store
+* Gather the results from the result set
```

Comment:
FindOne and MustFindOne returns no result set but an object, or an error, or a panic